sktime/clustering/spatio_temporal/_st_dbscan.py

```diff
@@ -0,0 +1,296 @@
+"""
+ST-DBSCAN - fast scalable implementation of ST DBSCAN
+            scales also to memory by splitting into frames
+            and merging the clusters together
+"""
+
+__author__ = ["VectorNd"]
+
+from sktime.clustering import BaseClusterer
+
+import numpy as np
+
+from scipy.spatial.distance import pdist, squareform
+from scipy.sparse import coo_matrix
+
+from sklearn.cluster import DBSCAN
+from sklearn.utils import check_array
+from sklearn.neighbors import NearestNeighbors
+
+import warnings
+
+class ST_DBSCAN(BaseClusterer):
+    """
+    A class to perform the ST_DBSCAN clustering
+    Parameters
```

Comment:
newlines are missing in the docstrings, e.g., before section headers - ensure to conform with the code formatting requirements. Here's how you can set them up on your computer, automatically: https://www.sktime.net/en/latest/developer_guide/coding_standards.html